xdsl/transforms/convert_riscv_to_llvm.py

```diff
@@ -0,0 +1,157 @@
+from dataclasses import dataclass
+
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, riscv
+from xdsl.dialects.builtin import IntAttr, IntegerAttr, UnrealizedConversionCastOp
+from xdsl.dialects.llvm import InlineAsmOp
+from xdsl.dialects.riscv import IntRegisterType, RISCVInstruction
+from xdsl.ir import Operation, OpResult, SSAValue
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.traits import HasInsnRepresentation
+from xdsl.utils.exceptions import DiagnosticException
+
+
+@dataclass(frozen=True)
+class RiscvToLLVMPattern(RewritePattern):
+    xlen: int
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: RISCVInstruction, rewriter: PatternRewriter):
+        ops_to_insert: list[Operation] = []
+
+        # inputs for the llvm inline asm op
+        assembly_args_str: list[str] = []
+        constraints: list[str] = []
+        inputs: list[SSAValue | OpResult] = []
+        # number of results produced from the inline assembly op
+        # all results are considered to be XLEN long integers
+        num_results: int = 0
+        # keep track which results are taken from the inline assembly op and which one are "other":
+        #  int      -> index into the inline asm op result list
+        #  ssa val  -> use this ssa value instead (e.g. when the op "returns" the zero register)
+        result_map: list[int | SSAValue] = []
+
+        # populate assembly_args_str and constraints
+        for arg in op.assembly_line_args():
+            # ssa value used as an output operand
+            if (
+                isinstance(arg, OpResult)
+                and isinstance(arg.type, IntRegisterType)
+                and arg.op is op
+            ):
```

Comment:
could you please use match syntax here?

Reply:
Also, I think `arg.owner is op` is enough to determine whether an assembly arg is an operand or result